module/mempool/herocache/backdata/heropool/pool.go

```diff
@@ -159,22 +163,34 @@ func (p Pool) Head() (flow.Entity, bool) {
 // Ejection happens if there is no available slot, and there is an ejection mode set.
 // If an ejection occurred, ejectedEntity holds the ejected entity.
 func (p *Pool) sliceIndexForEntity() (i EIndex, hasAvailableSlot bool, ejectedEntity flow.Entity) {
+	lruEject := func() (EIndex, bool, flow.Entity) {
+		// LRU ejection
+		// the used head is the oldest entity, so we turn the used head to a free head here.
+		invalidatedEntity := p.invalidateUsedHead()
+		return p.claimFreeHead(), true, invalidatedEntity
+	}
+
 	if p.free.head.isUndefined() {
 		// the free list is empty, so we are out of space, and we need to eject.
 		switch p.ejectionMode {
 		case NoEjection:
 			// pool is set for no ejection, hence, no slice index is selected, abort immediately.
 			return 0, false, nil
-		case LRUEjection:
-			// LRU ejection
-			// the used head is the oldest entity, so we turn the used head to a free head here.
-			invalidatedEntity := p.invalidateUsedHead()
-			return p.claimFreeHead(), true, invalidatedEntity
 		case RandomEjection:
 			// we only eject randomly when the pool is full and random ejection is on.
-			randomIndex := EIndex(rand.Uint32() % p.size)
+			random, err := rand.Uint32n(p.size)
+			if err != nil {
+				p.logger.Warn().Err(err).
```

Comment:
In this case lru is probably fine, but should a lack of randomness crash the node?
```suggestion
				p.logger.Fatal().Err(err).
```

Reply:
Does a `Fatal()` or `Panic()` log cause the node to crash eventually?  or does that depend on the upper layers receiving the logs.

Btw, if the `random` package fails to return randomness, the node software would have other problems, components where the error is handled properly will crash the node.

Reply:
> Does a `Fatal()` or `Panic()` log cause the node to crash eventually? or does that depend on the upper layers receiving the logs.

yes

Reply:
https://github.com/onflow/flow-go/pull/4052/commits/183ce49f4019b910422809b924258923cb49f114